cli/src/commands/deploy.js

```diff
@@ -0,0 +1,186 @@
+const { reporter, chalk } = require('@dhis2/cli-helpers-engine')
+const path = require('path')
+const fs = require('fs-extra')
+const FormData = require('form-data')
+const inquirer = require('inquirer')
+
+const makePaths = require('../lib/paths')
+const parseConfig = require('../lib/parseConfig')
+const { createClient } = require('../lib/httpClient')
+
+const constructAppUrl = (baseUrl, config, serverVersion) =>
```

Comment:
would be nice with some comments/docs for these long nested ternary blocks of code to make it more understandable , or split up to a normal block, I find this quite hard to read now without really focusing and putting some time in it.

Might be enough with some jsdoc or 2-3 example outputs of the url depending on the version <35 or >= 35

Reply:
Updated (moved to a separate file, got rid of the complex ternary, added comments to clarify why behavior is different) let me know if it's easier to understand!

Reply:
https://github.com/dhis2/app-platform/pull/451/files#diff-0fac56a67ec46d47cec4c17ae9c1db6b